README.rst
=====
IMS
=====

IMS is a simple Django app to for managing inventory.  IMS stands for
Inventory Management System.  The base models are:
Site - location where inventory is stored
ProductInformation - description of inventory types
InventoryItem - record recording the status of a product at a site

Detailed documentation is in the "docs" directory.

Quick start
-----------

1. Add "ims" to your INSTALLED_APPS setting like this::

    INSTALLED_APPS = (
        ...
        'ims',
    )

2. Include the ims URLconf in your project urls.py like this::

    url(r'^ims/', include('ims.urls')),

3. Run `python manage.py migrate` to create the ims models.

4. Start the development server and visit http://127.0.0.1:8000/admin/
   to administer the ims models (you'll need the Admin app enabled).

5. Visit http://127.0.0.1:8000/ims/ to work with inventory.

6. To run tests:
   replace the dummy .secret_file.json in the ims_tests package with a file
   containing your Django environment secret information.
   The default file "ims_tests/.secret_file.json" looks like this:
   {
       "IMS_SECRET":"secret_key",
       "IMS_DB":"dbName",
       "IMS_DB_USER":"dbUserName",
       "IMS_DB_PASS":"dbPassword"
   }

7. Be careful with the .secret_file.json, once you have populated it with your
   secret environment variables.  You don't want unauthorized users to be able
   to read them.

8. to run the tests in the ims_tests package dir:
   python run_tests.py


docs/RELEASE_NOTES.txt
2019/09/12 Version 1.19.9
fix: change from http to https for jquery code load to avoid 'Blocked loading mixed active content' error
2018/06/24 Version 1.19.8
feat: Added cost per item to site inventory, inventory detail and inventory status reports. Require mysqlclient==1.3.12.
2016/02/17 Verion 1.19.7:
fix: Set modified when changing ProductInformation. Handle unicode coming in from spreadsheet restores. Reverse order of RELEASE_NOTES and __init__.py version tuples.. Change date format of __init__.py version info
2016/01/23 Version 1.19.6:
fix: Fix check of inventory when deleting products, so deleted inventory is ignored
2016/01/23 Version 1.19.5:
* ce97188 (HEAD, master) fix: Add @never_cache decorator to all views to override W3TC plugin .htaccess overrides
2016/01/22 Version 1.19.4:
* f51de1e (HEAD, master) feat: Skip testing of HomeView since we no longer use IMS page
2016/01/21 Version 1.19.3:
* e801151 (HEAD, master) style: Enhance styling of help text and field spacing in detail forms for site and product
2016/01/21 Version 1.19.2:
* bdfbcbc (HEAD, master) fix: Calculation of ProductInformation.num_sites_containing
2016/01/19 Version 1.19.1:
* 7e5f581 (HEAD, master) refactor: Add lots of new tests and refactor to enhance test
* e0c9fe3 refactor: Add view for adding a single product to site inventory. Split out site_delete to a stand-alon
* 2f6769e feat: add new urls for adding a product directly to a site from product_details page
* 511fa4b feat: Add template for selecting a site to which to add a product
* f4a1f59 feat: Add new Add to Site button on product_detail page, so we can go directly to adding inventory to a
* dcbbf78 refactor: create separate templates for adding a single product and multiple products
* fafbda3 refactor: Separate product_detail and product_add templates
* 7efdcef feat: re-arrange info, warning, and error announcements, so they appear in that order
* 4b0bb1f fix: generalize readonly styling for input, select, textarea
* 05185d9 feat: add readonly form for product_select_add_site view and inventoryItemNoSiteNoDelete for product_ad
* bf57d84 fix: product_add view redirected with page view param, which was deprecated.  Also handle product_detail * 7e5f581 (HEAD, master) refactor: Add lots of new tests and refactor to enhance test
* e0c9fe3 refactor: Add view for adding a single product to site inventory. Split out site_delete to a stand-alon
* 2f6769e feat: add new urls for adding a product directly to a site from product_details page
* 511fa4b feat: Add template for selecting a site to which to add a product
* f4a1f59 feat: Add new Add to Site button on product_detail page, so we can go directly to adding inventory to a
* dcbbf78 refactor: create separate templates for adding a single product and multiple products
* fafbda3 refactor: Separate product_detail and product_add templates
* 7efdcef feat: re-arrange info, warning, and error announcements, so they appear in that order
* 4b0bb1f fix: generalize readonly styling for input, select, textarea
* 05185d9 feat: add readonly form for product_select_add_site view and inventoryItemNoSiteNoDelete for product_ad
* bf57d84 fix: product_add view redirected with page view param, which was deprecated.  Also handle product_detai* 7e5f581 (HEAD, master) refactor: Add lots of new tests and refactor to enhance test
* e0c9fe3 refactor: Add view for adding a single product to site inventory. Split out site_delete to a stand-alon
* 2f6769e feat: add new urls for adding a product directly to a site from product_details page
* 511fa4b feat: Add template for selecting a site to which to add a product
* f4a1f59 feat: Add new Add to Site button on product_detail page, so we can go directly to adding inventory to a
* dcbbf78 refactor: create separate templates for adding a single product and multiple products
* fafbda3 refactor: Separate product_detail and product_add templates
* 7efdcef feat: re-arrange info, warning, and error announcements, so they appear in that order
* 4b0bb1f fix: generalize readonly styling for input, select, textarea
* 05185d9 feat: add readonly form for product_select_add_site view and inventoryItemNoSiteNoDelete for product_ad
* bf57d84 fix: product_add view redirected with page view param, which was deprecated.  Also handle product_detail view when there are no sites.
2015/12/27 Version 1.19.0:
* c27e259 (HEAD, tag: v1.19.0, master) Update to Django 1.9 from 1.7
2015/12/26 Version 1.18.0:
* c163100 (HEAD, master, filtering) Add new filtering features
2015/12/21 Version 1.17.0:
* 99df58b (HEAD, master) Add new sorting features
2015/12/14 Version 1.16.0:
* e744490 (pagination) refactor Update pagination to reference paginatedItems.  Refactor pagination methods
* eb28854 feat replace laquo and raquo with font-awesome.  replace page navigation with new pagination class based approach.
* 94a00d6 style unify page display style across site.  Also add pagination css.
* 5b74185 feat remove PAGE_SIZE from settings
* 3230ee0 feat use pagination class and session for paging now
* ef789e9 feat remove paging information frm URLs, use session and pagination now
* e84ae03 feat pass in pageSize information to changed inventory methods
* b14fc01 add font-awesome css to ims/static
* 93931c2 fix link to style.css in report print popup javascript function
* 4c6f0ec added category creation to tests
2015/11/22 Version 1.15.3:
* 248e289 (HEAD, master) handle backups without picture files
2015/11/22 Version 1.15.2:
* 485b72f (HEAD, master) fix restore of originalPictureName
2015/11/22 Version 1.15.1:
* 09e05ce (HEAD, master) fix backup of thumbnail file
* 19ee201 save ProductInformation.originalPictureName relative to MEDIA_ROOT instead of absolute path
2015/11/21 Version 1.15.0:
* f2190db (HEAD, master) enable restore of archive backups including images.  add error checking for missing picture files
2015/11/15 Version 1.14.1:
* fadf785 (HEAD, master) left justify category and code in products list

ims/__init__.py
VERSION=('1.19.9','2018/09/12','change from http to https for jquery code load to avoid "Blocked loading mixed active content" error')
#VERSION=('1.19.8','2018/06/24','feat: Added cost per item to site inventory, inventory detail and inventory status reports. Require mysqlclient==1.3.12.')
#VERSION=('1.19.7','2016/02/17','fix: Set modified when changing ProductInformation. Handle unicode coming in from spreadsheet restores. Reverse order of RELEASE_NOTES and __init__.py version tuples.. Change date format of __init__.py version info')
#VERSION=('1.19.6','2016/01/23','Fix check of inventory when deleting products, so deleted inventory is ignored')
#VERSION=('1.19.5','2016/01/23','Add @never_cache decorator to all views to override W3TC plugin .htaccess overrides')
#VERSION=('1.19.4','2016/01/22','Remove IMS page, its not used.  Redirect to Home if IMS page requested.')
#VERSION=('1.19.3','2016/01/21','Enhance styling of help text and field spacing in detail forms for site and product')
#VERSION=('1.19.2','2016/01/21','Fix calculation of ProductInformation.num_sites_containing')
#VERSION=('1.19.1','2016/01/19','Add lots of tests and refactor views to enhance test')
#VERSION=('1.19.0','2015/12/27','Update to Django 1.9 from 1.7')
#VERSION=('1.18.0','2015/12/26','Add new filtering features')
#VERSION=('1.17.0','2015/12/21','Add new sorting features')
#VERSION=('1.16.0','2015/12/14','New pagination using Pagination class')
#VERSION=('1.15.3','2015/11/22','handle backups without picture files')
#VERSION=('1.15.2','2015/11/22','fix restore of originalPictureName')
#VERSION=('1.15.1','2015/11/22','fix ProductInformation.originalPictureName path and thumbnail backup')
#VERSION=('1.15.0','2015/11/21','add category and picture export and import')
#VERSION=('1.14.1','2015/11/15','left justify category and code in products list')
#VERSION=('1.14.0','2015/11/15','updates to ProductCode and Category, mostly UI improvements')
#VERSION=('1.13.0','2015/12/11','make product code 36 char and auto generate uuid4 compliant codes in js')
#VERSION=('1.12.1','2015/12/11','fix generate code button and templates that use it')
#VERSION=('1.12.0','2015/10/11','allow auto generation of product codes. Add date and comments to version')
#VERSION='1.11.1'
#VERSION='1.11.0'